### PR TITLE
[#72] Fix Haskell JSON encoder for unary constructors

### LINKED ISSUE #72: Haskell json encoders incorrectly derived for data types with single constructor
```
data NewDataType
    = OnlyConstructor
    deriving stock (Show, Read, Eq, Generic)
    deriving (Elm, ToJSON, FromJSON) via ElmStreet ProjectType
```

This will produce the following problem with json encoding:

```
encode OnlyConstructor
> "[]"
```

The generated Elm encoder and decoder seem correct, but the generated Haskell encoder seems wrong. Adding a second constructor to the `NewDataType` fixes the issue.

```
data NewDataType
    = OnlyConstructor
    | AnotherConstructor
    deriving stock (Show, Read, Eq, Generic)
    deriving (Elm, ToJSON, FromJSON) via ElmStreet ProjectType
```
which now gives the correct encoding
```
encode OnlyConstructor
> "\"OnlyConstructor\""
```

---
This was checked on GHCi, version 8.6.3

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #72

### CODE REVIEW

File: src/Elm/Aeson.hs [hunk 1]
```diff
@@ -133,6 +133,7 @@ elmStreetToJson = genericToJSON (elmStreetJsonOptions @a)
 elmStreetJsonOptions :: forall a . Typeable a => Options
 elmStreetJsonOptions = defaultOptions
     { fieldLabelModifier = T.unpack . stripTypeNamePrefix typeName . T.pack
+    , tagSingleConstructors = True
```
Comment (vrom911) [inline src/Elm/Aeson.hs]: :open_mouth:

Comment (chshersh) [reply src/Elm/Aeson.hs]: Yeah, it has a lot of flags...

Comment (vrom911) [reply src/Elm/Aeson.hs]: Maybe it shouldn't be done when newtypes or constructors with fields? Or this is considered here?

Comment (chshersh) [reply src/Elm/Aeson.hs]: I hope that this option takes care of this. Anyway, for `newtype` we are using `newtype` strategy so it's not affected...

**Pull request merged**